provider/openstack/provider.go

```diff
@@ -1275,11 +1275,12 @@ func (e *Environ) startInstance(
 
 	server, err := tryStartNovaInstance(shortAttempt, e.nova(), opts)
 	if err != nil || server == nil {
-		err := errors.Annotate(err, "cannot run instance")
+		logger.Debugf("cannot run instance full error: %q", err)
+		errCause := errors.Cause(err)
+		err := errors.Annotate(errCause, "cannot run instance")
 		// Improve the error message if there is no valid network.
 		if isInvalidNetworkError(err) {
-			msg := "\n\tThis error was caused by juju attempting to create an OpenStack instance with no network defined.\n\tNo network has been configured, nor could juju find an internal OpenStack network to suggest."
-			err = errors.New(fmt.Sprintf("%s%s", err.Error(), msg))
+			err = errors.New(fmt.Sprintf("%s%s", err.Error(), e.invalidNetworkErrorMessageForUser()))
```

Comment:
This looks a bit strange... why not just pass `err` to `invalidNetworkErrorMessageForUser` (btw, perhaps `userFriendlyNetworkError` would be a more suitable name) and have it return an error back? Wouldn't just annotating the error with the user-friendly message do the trick?

Reply:
Updating the method name. 

Due to the length of the error, annotating it will make it harder to read now to to fix imo.